### PR TITLE
Adding tests for the utils module

### PR DESCRIPTION
**Pull request by rihp**
### PR Quality Checklist
- [x] My pull request is atomic and focuses on a single change.
- [ ] I have thoroughly tested my changes with multiple different prompts.
- [x] I have considered potential risks and mitigations for my changes.
- [x] I have documented my changes clearly and comprehensively.
- [x] I have not snuck in any "extra" small tweaks changes

### CODE REVIEW

**Comment by ntindle** [comment]
Run the linter to fix the formating

File: tests/test_utils.py [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+@patch("requests.get")
+def test_get_bulletin_from_web_failure(mock_get):
+    mock_get.return_value.status_code = 404
+    bulletin = get_bulletin_from_web()
+    print(bulletin)
+    assert bulletin == ""
+
+
+def test_get_current_git_branch():
+    branch_name = get_current_git_branch()
```
Comment (ntindle) [inline tests/test_utils.py:76]: This won’t work in ci as GitHub actions doesn’t check out the branch but the commit SHA

see docs here https://github.com/actions/checkout#usage

add a decorator to skip in CI to test/utils.py

Comment (ntindle) [reply tests/test_utils.py:76]: Do this but in tests/utils.py similar to how the existing one there is to skip if the api key doesn’t exist

Comment (rihp) [reply tests/test_utils.py:76]: this seems to be resolved

Comment (ntindle) [reply tests/test_utils.py:76]: Right, move the snippet you used to skip to the test/utils.py so other tests can use it as it’s super useful! There’s an example of how it’s done in that file

File: tests/utils.py [hunk 2]
```diff
@@ -16,3 +16,10 @@ def wrapper(*args, **kwargs):
         return wrapper
 
     return decorator
+
+
+def skip_on_ci(test_function):
+    return pytest.mark.skipif(
+        os.environ.get("CI") == "true",
+        reason="This test doesn't work on GitHub Actions.",
+    )(test_function)
```
Comment (ntindle) [inline tests/utils.py]:
```suggestion
def skip_in_ci():
    def decorator(func):
        def wrapper(*args, **kwargs):
            if os.environ.get("CI"):
                pytest.skip(f"This test doesn't work on GitHub Actions.")
            else:
                return func(*args, **kwargs)
        return wrapper
    return decorator
```

Comment (ntindle) [reply tests/utils.py]: give this a shot so the pattern is followed correctly. No idea if it'll work

**Review by ntindle — APPROVED** [review]
LGTM

**Pull request merged**